Products/ZenModel/migrate/addSolrService.py

```diff
@@ -0,0 +1,153 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import json
+import logging
+import os
+log = logging.getLogger("zen.migrate")
+
+import Migrate
+import servicemigration as sm
+sm.require("1.0.0")
+from servicemigration import HealthCheck
+
+
+class AddSolrService(Migrate.Step):
+    """
+    Add Solr service and associated healthchecks.
+    """
+    version = Migrate.Version(116, 0, 0)
+
+    def cutover(self, dmd):
+        try:
+            ctx = sm.ServiceContext()
+        except sm.ServiceMigrationError:
+            log.info("Couldn't generate service context, skipping.")
+            return
+
+        changed = False
+        # If the service lacks Solr, add it now.
+        solr = filter(lambda s: s.name == "Solr", ctx.services)
+        log.info("Found %i services named 'Solr'." % len(solr))
+        if not solr:
+            log.info("No Solr found; creating new service.")
+            new_solr = default_solr_service()
+            infrastructure = ctx.findServices('^[^/]+/Infrastructure$')[0]
+            ctx.deployService(json.dumps(new_solr), infrastructure)
+            changed = True
+
+        # Now healthchecks
+        solr_answering_healthcheck = HealthCheck(
+            name="solr_answering",
+            interval=10.0,
+            script="curl -A 'Solr answering healthcheck' -s http://localhost:8983/solr/zenoss_model/admin/ping?wt=json | grep -q '\"status\":\"OK\"'"
+        )
+
+        for svc in ctx.services:
+            if filter(lambda c: c.name == 'solr_answering', svc.healthChecks):
+                continue
+            for ep in svc.endpoints:
+                if ep.purpose == 'import' and ep.application == 'zodb_.*':
+                    svc.healthChecks.append(solr_answering_healthcheck)
+                    changed = True
+                    break
+
+        if changed:
+            ctx.commit()
+
+
+def default_solr_service():
+    return {
+    "CPUCommitment": 2,
+    "Command": "setuser zenoss /opt/solr/zenoss/bin/start-solr -cloud -Dbootstrap_confdir=/opt/solr/server/solr/configsets/zenoss_model/conf -Dcollection.configName=zenoss_model",
```

Comment:
Can we do that without knowing the value yet? We can update the migrate script once that value is determined, I think.